### PR TITLE
Looser coupling between HUD API and code in cfgov/legacy/views/

### PR DESCRIPTION
**Pull request by rosskarchner**
## Additions

- DJANGO_HUD_API_ENDPOINT setting, which can be populated from the
environment. The default matches what we use in production.
- a useful initial value for development is added to .env_SAMPLE

## Changes

- the new setting is used to connect to the HUD API. The previous code had hardcoded "http://localhost" which doesn't work when developing on localhost, and depended on Django's URL reversing, which wouldn't work in a hypothetical future where the HUD API was running as a standalone service.

## Testing

- After adding something like `export HUD_API_ENDPOINT=http://localhost:8000/hud-api-replace/` to your own .env file, (and re-sourcing it), you should be able to successfully search for zip codes at /find-a-housing-counselor/

## Checklist

* [x] PR has an informative and human-readable title
* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [x] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: cfgov/legacy/views/housing_counselor.py [hunk 1]
```diff
@@ -40,14 +41,10 @@ def get_context_data(self, **kwargs):
...
         """Return list of housing counselors closest to a given zipcode.
 
-        Queries a locally running django-hud API.
+        Queries a local or remote django-hud API
         """
-        api_path = reverse(
-            'hud_api_replace:index',
-            kwargs={'zipcode': zipcode}
-        )
+        api_url = urljoin(settings.DJANGO_HUD_API_ENDPOINT,zipcode)
```
Comment (chosak) [inline cfgov/legacy/views/housing_counselor.py]: This might fail flake8.

File: cfgov/legacy/views/housing_counselor.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import re
 import requests
 import sys
+from urlparse import urljoin
```
Comment (chosak) [inline cfgov/legacy/views/housing_counselor.py]: Would you mind alphabetizing this with the other third-party imports below (under `django.views.generic`)?

Comment (rosskarchner) [inline cfgov/legacy/views/housing_counselor.py]: Even though it's in the standard library?

Comment (chosak) [inline cfgov/legacy/views/housing_counselor.py]: Oh, fair point. I withdraw.

**Pull request merged**